rcongui_public/src/pages/games/horizontal-games-list.tsx

```diff
@@ -1,41 +1,56 @@
 import {ScoreboardMap, ScoreboardMaps} from "@/types/api";
 import {Link, useLocation} from "react-router";
-import React, {useEffect, useRef, useState} from "react";
+import React, {useEffect, useMemo, useRef, useState} from "react";
 import {ScrollArea, ScrollBar} from "@/components/ui/scroll-area";
 import {validGame} from "@/pages/games/list";
 import MapFigure from "@/components/game/map-figure";
-import dayjs from "dayjs";
+import dayjs, {Dayjs} from "dayjs";
 import {cn, dayjsLocal} from "@/lib/utils";
 import {useTranslation} from "react-i18next";
 import {useLocale} from "@/i18n/locale-provider";
 
+const pathRegex = (gameId: number) => new RegExp(`^/games/${gameId}(?:/|$)`);
+
 export const HorizontalGamesList = ({ games }: { games: ScoreboardMaps }) => {
   const { pathname } = useLocation()
-  const [hiddenDates, setHiddenDates] = useState<Record<string, boolean>>({});
+  const [gamesVisibility, setGamesVisibility] = useState<Record<number, boolean>>({});
+  const gamesVisibilityRef = useRef(gamesVisibility);
```

Comment:
for what is this actually needed? Wouldn't `gamesVisibility` be enough in all cases where the ref is currently used?